tapgarden/planter.go

```diff
@@ -357,39 +358,74 @@ func (c *ChainPlanter) Start() error {
 				batch.AssetMetas = make(AssetMetas)
 			}
 
+			// If batch funding or sealing fail during startup, the
+			// batch will be marked as cancelled. The batch can
+			// still be displayed by the planter, and can be
+			// resubmitted manually.
+			cancelBatch := func() {
+				log.Warnf("Marking batch as cancelled (%x)",
+					batchKey)
+				err := c.cfg.Log.UpdateBatchState(
+					ctx, batch.BatchKey.PubKey,
+					BatchStateSeedlingCancelled,
+				)
+
+				// If updating the batch state fails, the batch
+				// will still be skipped on this startup; we can
+				// continue without passing the error further.
+				if err != nil {
+					log.Warnf("Unable to cancel batch (%x)",
+						batchKey)
+				}
+			}
+
 			// TODO(jhb): Log manual fee rates?
 			// If the batch was still pending, or if batch
 			// finalization was interrupted, it may need to be
 			// funded or sealed before being assigned a caretaker.
 			// A batch that was already properly frozen at this
 			// point should not be modified before being assigned a
 			// caretaker.
-			batchKey := batch.BatchKey.PubKey.SerializeCompressed()
 			if batchState == BatchStatePending ||
 				batchState == BatchStateFrozen {
 
+				var (
+					fundErr error
+					sealErr error
+				)
+
 				if !batch.IsFunded() {
 					log.Infof("Funding non-finalized "+
 						"batch from DB (%x)", batchKey)
-					err := c.fundBatch(
+					fundErr = c.fundBatch(
 						ctx, FundParams{}, batch,
 					)
-					if err != nil {
-						startErr = err
-						return
-					}
+				}
+
+				if fundErr != nil {
+					log.Warnf("Failed to fund batch from "+
+						"DB (%x): %s",
+						batchKey, fundErr.Error())
+					cancelBatch()
+					break
```

Comment:
Good catch! I think right now we shouldn't be able to end up with multiple non-finalized batches on disk, since we only handle one batch at a time. But will fix + update tests.